image.go

```diff
@@ -23,6 +23,7 @@ import (
 	"github.com/containerd/containerd/content"
 	"github.com/containerd/containerd/errdefs"
 	"github.com/containerd/containerd/images"
+	"github.com/containerd/containerd/images/encryption"
```

Comment:
This is the import which brought it in all encryption into the client

Reply:
What do you suggest? Get rid of the functionality?

Reply:
Once an image has been pulled, the `sn.Stat()` in `rootfs/apply.go::ApplyLayer()` does not return an error anymore and `applyLayers()` doesn't get called and we don't call into the daemon. So that's why wd did this it on the client side.

Reply:
Does the applier have access to the snapshotter so that it could call the `sn.Stat()`? We would have to either find out in `Apply()` that the unpacking already happened and don't do it again but only check for authorization or we pass this information into the daemon via a parameter [so that we only run the check for authorization then].

Reply:
```
               } else {
                       // The cached image is served, so we need to do an image authorization check.
                       // Image authorization check prevents access to the cached encrypted images
                       // without valid private keys that can decrypt those encrypted images.
                       if encryption.IsEncryptedDiff(ctx, layer.Blob.MediaType) {
                               err = encryption.CheckAuthorization(ctx, cs, i.Target(), iopts.decryptconfig)
                               if err != nil {
                                       return err
                               }
                       }
                }
```
We don't have `i.Target()` and its Descriptor on the server side in the `Apply()` function. So if we wanted to move this check out of the client we may need to come up with a new server side function?

Reply:
@dmcgowan the operation actually can be done in Apply() but I am not sure whether it should be done there since it's not going through the unpacking but cutting this short after checking authorization. 

I added two more patches on top of the existing ones that at least push this functionality to the server but still call `encryption.IsEncryptedDiff` on the client, which seems to pull the crypto packages in. This function can be easily moved but other things may need to be cleaned up, if this is the right direction.

Reply:
... the other option is to remove the above code from this PR and bring it back in the next one where we may want similar functionality in the paths of `container create` and `run`.

Reply:
If the flow here is good, I am wondering if `CheckAuthorizationOnly` as another option would make the common case less verbose.

Reply:
For the sake of simplifying this PR, I would almost tend to remove this functionality here ... I feel the problem is now moving into other parts of the code as well and it doesn't feel right to do this in `Apply()`. It feels like we are overloading this function with different functionality. Also, we may run into similar issues in the next PR where we may again need to push this into the daemon and maybe there should be a distinct API function to just check the authorization.